spec/notifications/client/precompiled_letter_spec.rb

```diff
@@ -37,18 +37,26 @@
         with(body: { reference: "12345", content: encoded_content })
     end
 
-    it "returns a ResponseNotification with an id and reference" do
+    it "hits the correct API endpoint with postage" do
+      input_string = StringIO.new("My precompiled letter")
+      encoded_content = Base64.strict_encode64(input_string.read)
+      input_string.rewind
+
+      client.send_precompiled_letter('12345', input_string, 'first')
+
+      expect(WebMock).to have_requested(:post, "https://#{uri.host}/v2/notifications/letter").
+        with(body: { reference: "12345", content: encoded_content, postage: 'first' })
+    end
+
+    it "returns a ResponsePrecompiledLetter with an id, postage and reference" do
       response = File.open('spec/test_files/test_pdf.pdf', 'rb') do |file|
-        client.send_precompiled_letter('my letter', file)
+        client.send_precompiled_letter('my letter', file, 'first')
```

Comment:
We don't really have a test that covers the case for if someone doesn't provide postage. Since the test above is testing for postage being 'first', you could change this test to not pass in postage at all, then expect the postage that's returned to be 'second'

Reply:
but isn't response coming from a mock in this test?

Reply:
Yeah, you're right - ignore that suggestion. I was forgetting that the default of 'second' was coming from api, not here.